### PR TITLE
gopass: update to 1.12.0.

### PR DESCRIPTION
**Pull request by dshynkev**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by dshynkev** [comment]
One thing of note: this version changed the way the `gpg` binary is selected. They assume `gpg` points to the version of GPG the user uses, but on Void it points to GPG v1, which is probably not what most use.

One must `export GOPASS_GPG_BINARY=$(which gpg)` for it to keep working.

Should I note this in a post-install message?

**Comment by ericonr** [comment]
I'd use `/usr/bin/gpg2` directly instead of which.

You could also open an RFC for dropping old GPG, if you think it's feasible.

**Comment by dshynkev** [comment]
It should be possible. A few of current gnupg revdeps definitely support gnupg2.

I'll file an RFC, but perhaps for now a post-install message is best?
It might take a while to phase out v1 if at least one of its revdeps really requires it.

I could also patch gopass if that's more appropriate. Sorry, not sure about the Void packaging etiquette.

**Comment by dshynkev** [comment]
I've tentatively added a message.

**Comment by ericonr** [comment]
> It might take a while to phase out v1 if at least one of its revdeps really requires it.

Fair enough :)

Ideally, we want to make the package with as little friction as possible. If Void still defaulting to gnupg 1 makes it an issue (assuming most users do actually use gnupg2 instead), we can consider patching, IMO. I'm tagging @fnogcps so they can leave their thoughts here.

**Comment by ahesford** [comment]
Upstream is doing this wrong. GPG2 provides `gpgconf` to provide a machine-readable way to discover the path to the gpg binary. If `gpgconf` is available, you query that for the full path to the `gpg` executable, which in Void returns `/usr/bin/gpg2`. If `gpgconf` is not available, then you fall back to looking for `gpg` in the path.

**Comment by ericonr** [comment]
@dshynkev see https://github.com/void-linux/void-packages/pull/28217 :)

File: srcpkgs/gopass/template [hunk 1]
```diff
@@ -13,7 +13,7 @@ license="MIT"
 homepage="https://www.gopass.pw/"
 changelog="https://raw.githubusercontent.com/gopasspw/gopass/master/CHANGELOG.md"
 distfiles="https://github.com/gopasspw/gopass/archive/v${version}.tar.gz"
-checksum=f4bfe03bc26701baf411795faafe774aa6025bc6050ab3ebf7e931e5ff821961
+checksum=26ac90ef60d7f08e7d9361280d6457edb90642e821edba2c5e5cb0a09ca20b08
 
 if [ "$CROSS_BUILD" ]; then
```
Comment (ericonr) [inline srcpkgs/gopass/template:18]: You should be able to remove this block if you switch to using `vtargetrun`, and the lines below could also use `vcompletion`. Please see `rustup` for an example.

Comment (dshynkev) [reply srcpkgs/gopass/template]: Got it, thanks!

**Comment by ahesford** [comment]
FYI, upstream is looking at a [properly looking for the `gpg2` binary](https://github.com/gopasspw/gopass/pull/1758). Might want to keep an eye on that PR and, if it's accepted, backport the patch for this release if they don't cut a new release.

**Comment by dshynkev** [comment]
Thanks for reporting it to them!

**Comment by ahesford** [comment]
https://github.com/gopasspw/gopass/commit/2b2c605b584cbfa6d56167f4b492c0db7698b4ac

**Comment by dshynkev** [comment]
The next version got tagged. Didn't want to carry a non-trivial patch (just that commit alone didn't work) when release was imminent, and so finally we're here. Sorry about the delay!

**Pull request merged**